src/components/application_manager/src/rpc_passing_handler.cc

```diff
@@ -0,0 +1,326 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "application_manager/help_prompt_manager_impl.h"
+
+#include <algorithm>
+#include <iterator>
+
+#include "application_manager/application.h"
+#include "application_manager/application_manager.h"
+#include "application_manager/rpc_passing_handler.h"
+#include "application_manager/commands/command_impl.h"
+#include "application_manager/message_helper.h"
+#include "application_manager/smart_object_keys.h"
+#include "encryption/hashing.h"
+#include "resumption/last_state.h"
+#include "smart_objects/enum_schema_item.h"
+#include "utils/logger.h"
+
+#include "smart_objects/enum_schema_item.h"
+#include "utils/timer_task_impl.h"
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "AppServiceManager")
+
+namespace application_manager {
+
+RPCPassingHandler::RPCPassingHandler(AppServiceManager& asm_ref,
+                                     ApplicationManager& am_ref)
+    : app_service_manager_(asm_ref), app_manager_(am_ref) {}
+
+bool RPCPassingHandler::UsingAppServices(uint32_t correlation_id) {
+  if (messages_using_core_.find(correlation_id) == messages_using_core_.end()) {
+    return true;
+  }
+  messages_using_core_.erase(correlation_id);
+  return false;
+}
+
+bool RPCPassingHandler::CanUseRPCPassing(int32_t function_id) {
+  auto services = app_service_manager_.GetActiveServices();
+  for (auto it = services.begin(); it != services.end(); ++it) {
+    auto handled_rpcs =
+        it->record[strings::service_manifest][strings::handled_rpcs];
+    for (size_t i = 0; i < handled_rpcs.length(); i++) {
+      if (handled_rpcs[i].asInt() == function_id) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+bool RPCPassingHandler::RPCPassThrough(smart_objects::SmartObject rpc_message) {
+  uint32_t origin_connection_key, correlation_id;
+  int32_t function_id, message_type;
+
+  origin_connection_key =
+      rpc_message[strings::params][strings::connection_key].asUInt();
+  correlation_id =
+      rpc_message[strings::params][strings::correlation_id].asUInt();
+  function_id = rpc_message[strings::params][strings::function_id].asInt();
+  message_type = rpc_message[strings::params][strings::message_type].asInt();
+
+  LOG4CXX_DEBUG(logger_, "RPC_PASSING: ");
```

Comment:
@ShobhitAd  nice to use traces instead of debug logging entrance to function

Reply:
Resolved in https://github.com/smartdevicelink/sdl_core/pull/2821/commits/985569c8ac930745a0f3aa5105c19f8de82de058